Update NAS2D dependency

DanRStevens opened the pull request:
Update NAS2D dependency, and CircleCI Docker image used for builds. Target C++17 standard.

ldicker83 commented:
There are conflicts with the makefile. I'm not familiar enough with makefile's to feel comfortable making changes to resolve the conflicts.

DanRStevens commented:
Ahh, fixed the conflict. I had changed the target standard from `c++11` to `c++17`, which was needed by one of the changes in this branch. Meanwhile, in another branch that was recently merged, I'd also updated the target standard from `c++11` to `c++17`, but had also renamed the variable defined on that line from `CFLAGS` (used for C code) to `CXXFLAGS` (used for C++ code). Hence the conflict. If both lines had been changed identically, Git can usually merge those without complaint.

ldicker83 commented:
Sounds good. Thanks!

Looks good, going to merge.
